settings.gradle.kts

```diff
@@ -1,12 +1,84 @@
+                maven(url = uri(additionalRepositoryProperty.get()))
+                logger.info("A custom Kotlin repository ${additionalRepositoryProperty.get()} was added")
+            }
+
+            /*
+             * This property group is used to build kotlinx.atomicfu against Kotlin compiler snapshots.
+             * When build_snapshot_train is set to true, kotlin_version property is overridden with kotlin_snapshot_version.
+             * Additionally, mavenLocal and Sonatype snapshots are added to repository list
+             * (the former is required for AFU and public, the latter is required for compiler snapshots).
+             * DO NOT change the name of these properties without adapting kotlinx.train build chain.
+             */
+            val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+            if (buildSnapshotTrainGradleProperty.isPresent) {
+                maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+            }
+
+        }
+    }
+}
 
 dependencyResolutionManagement {
+    val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+
+    @Suppress("UnstableApiUsage")
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
+
+    @Suppress("UnstableApiUsage")
+    repositories {
+
+        // TODO(Dmitrii Krasnov): maybe could be passed via additionalRepositoryProperty
+        maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev")
+
+        if (buildSnapshotTrainGradleProperty.isPresent) {
+            maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+        }
+
+        val additionalRepositoryProperty = providers.gradleProperty("community.project.kotlin.repo")
+            .orElse(providers.gradleProperty("kotlin_repo_url"))
+        if (additionalRepositoryProperty.isPresent) {
+            maven(url = uri(additionalRepositoryProperty.get()))
+            logger.info("A custom Kotlin repository ${additionalRepositoryProperty.get()} was added")
+        }
+
+        mavenCentral()
+
+        exclusiveContent {
+            forRepository {
+                ivy("https://nodejs.org/download/v8-canary") {
+                    patternLayout { artifact("v[revision]/[artifact](-v[revision]-[classifier]).[ext]") }
+                    metadataSources { artifact() }
+                    content { includeModule("org.nodejs", "node") }
+                }
+            }
+            filter { includeGroup("org.nodejs") }
+        }
+
+        exclusiveContent {
+            forRepository {
+                ivy("https://github.com/yarnpkg/yarn/releases/download") {
+                    patternLayout { artifact("v[revision]/[artifact](-v[revision]).[ext]") }
+                    metadataSources { artifact() }
+                    content { includeModule("com.yarnpkg", "yarn") }
+                }
+            }
+            filter { includeGroup("com.yarnpkg") }
+        }
```

Comment:
Is there a YouTrack ticket to make manually adding the node/yarn repos unnecessary? There's https://youtrack.jetbrains.com/issue/KT-51379, but that's for the K/N repos.

Reply:
I see this task: https://youtrack.jetbrains.com/issue/KT-55620/

Reply:
And I've asked Ilya about current status

Reply:
UPD: @adam-enko Starting from the `2.0.0` we will be able to pass `null` repository URL to plugin and then use even `RepositoriesMode.FAIL_ON_PROJECT_REPOS`, but we will need to declare these repositories anyway

Reply:
UPD: there is also https://youtrack.jetbrains.com/issue/KT-34732, but it is only in backlog

Reply:
Put a comment